### ISSUE TITLE
sh: 1: mkfs.: not found when installing over a previously encrypted partition

### ISSUE THREAD

**Issue by PilotGuy772** [Vanilla-OS/Albius#80]
I was installing VanillaOS with an encrypted home partition and all was well until I ran into the issue described in #77. I rebooted to try again, this time without swap. I remembered that I have a SED, so I don't need an encrypted home partition, and I continued the installation without choosing encryption. I was using manual disk setup to install alongside Windows, too, but I'm not sure it matters. 

I used all the same partitions that were setup previously. I got an error very shortly into installation during the setup stage that went something like `sh: 1: mkfs.: not found`. I was tearing my hair out for a **while** until I figured to just create the ext4 filesystem myself in the terminal. It turns out that the crypt format was still there and mkfs.ext4 asked for confirmation to replace the crypt format with ext4. I obliged and restarted the installation, and the setup proceeded normally. I suppose you might want to investigate /core/recipe.go line 270. 

Anyway, this seems like it's certainly a bug. The error message is very strange and is consistent with what you would see if you just run `mkfs.` in bourne shell... I'm not really sure what's up with that. This also seems both relatively low priority (I can't imagine a lot of people are going to be in this situation) and relatively easy to patch.

**Comment by SpanishHans** [comment]
Can second this. Var fails to be created if previous install was also encrypted. Temporary fix seems to be a full drive quickwipe.

**Comment by porteusconf** [comment]
Me 3! Same error "...`mkfs. :not found`..." using VanillaOS2 live installer. The error messages indicate `%s` is empty in 
```
core/disk/filesystem.go:		makefsCmd := "mkfs.%s -F %s"
core/disk/filesystem.go:		makefsCmd := "mkfs.%s -f %s"
```
I can reproducibly get the error, even if I set each of the 4 partitions format to 'cleared' (no file-system) in parted/gparted, so I'm not sure the cause is just the partition being previously formatted (tho I did see lvcreate errors on subsequent attempts). I think the `%s argument is not assigned correctly. In my case the log says unable to format sda19, though the excerpt below does have "ext4" param:
```
  "setup" : [
      {
         "disk" : "/dev/sda",
         "operation" : "format",
         "params" : [
            "19",
            "ext4",
            "vos-boot"
         ]
      },
      {
         "disk" : "/dev/sda",
         "operation" : "format",
         "params" : [
            "20",
            "fat32",
            "vos-efi"
         ]
      },
```

I  can provide the both json files found at  /tmp/t*.json with install settings if you want, along output of parted  /dev/sda print, and the log file with install errors  if you tell me where to  find it  ;-)  I am also trying a dual or tripple boot on lenovo laptop, and all the other linux flavors (debian, even chromeos-flex) run flawlessly. Hardware tests of drive  and ram find no issues, either.

P.S. The first time I tried the install, I chose manual partitoning with gparted and it somehow finished install  without  error even tho each of the 4 empty partitions I created were formatted to ext4 in gparted. I think there was a post-install error, so it booted to blank screen even with refind boot manager able to find it. If I reproduce the boot fail, I'll  submit an issue for boot-failure. But every one of the many subsequent times I boot the live-iso and run installer, it fails in a few seconds after the install begins,  with the `mkfs. not found` in the log. Is there any alternative installer method? Text-only is ok. I'm willing to manually edit the json installer settings.

Other installers for other flavors of linux  usually, for most partitions, let you choose to format the partition, or leave the existing format (ext4 for example). I am wondering if tthat might be a worthwhile feature request as it might provide a work-around if albius is having problems doing mkfs...
